rdr_service/api/ppsc_intake_api.py

```diff
@@ -86,6 +86,17 @@ def validate_payload(self, *, req_data: dict):
         else:
             raise BadRequest("No activity_date_time_value provided.")
 
+        # Check for Primary Consent
+        if req_data['eventType'] != "Primary Consent":
+            with self.dao.session() as session:
+                is_primary = session.query(ConsentEvent).filter(
+                    ConsentEvent.participant_id == req_data['participantId'].split('P')[1],
+                    ConsentEvent.event_type_name == 'Primary Consent'
```

Comment:
Can you add two more conditions?
`ConsentEvent.data_element_name == 'activity_status'`
and 
`ConsentEvent.data_element_value == 'yes'`

Also, can you put this logic into its own method?

Reply:
Also, if this is in its own method, we might want to parameterize the `event_type_name`, `data_element_name` and `data_element_value` so we can check other consents if the need arised, i.e. EHR, DICOM, etc.